dashboard/src/modules/components/ComparisonComponent/PanelContent.jsx

```diff
@@ -33,39 +33,40 @@ const PanelConent = () => {
     [publicData, onFilter]
   );
   const [page, setPage] = useState(CONSTANTS.START_PAGE_NUMBER);
+
   return (
     <>
       {filteredDatasets.length > 0 && (
         <div className="datasets-container">
           {isCompareSwitchChecked ? (
             <div className="dataset-list-checkbox">
-              {filteredDatasets.map((item) => (
-                <Checkbox
-                  key={item.resource_id}
-                  label={item.name}
-                  id={item.resource_id}
-                  isChecked={selectedResourceIds?.includes(item.resource_id)}
-                  onChange={(checked) =>
-                    dispatch(setSelectedId(checked, item.resource_id))
-                  }
-                />
-              ))}
+              {filteredDatasets.map((item) =>
+                item ? (
```

Comment:
@dbutenhof, what was the answer to your questions??

Reply:
The filter step results in a sparse array, so there *can* be null entries. In the scope of this fix, I left it at that rather than suggesting a deeper refactoring, whether or not that might prove practical.

Reply:
Oh, dear.  [`Arrary.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) is not supposed to produce a sparse array (in fact, it skips over empty slots in the input array).  So, I wonder how we're ending up with empty slots....  😢

Reply:
The `filter` function's callback is returning items even if it is `undefined`. That's the reason for sparse array. Now, addressed the issue.